applications/GeoMechanicsApplication/custom_constitutive/small_strain_umat_law.cpp

```diff
@@ -643,24 +690,26 @@ Matrix& SmallStrainUMAT3DLaw::CalculateValue(ConstitutiveLaw::Parameters& rParam
     return rValue;
 }
 
-Vector& SmallStrainUMAT3DLaw::GetValue(const Variable<Vector>& rThisVariable, Vector& rValue)
+template <unsigned int TVoigtSize>
+Vector& SmallStrainUMATLaw<TVoigtSize>::GetValue(const Variable<Vector>& rThisVariable, Vector& rValue)
 {
     if (rThisVariable == STATE_VARIABLES) {
         if (rValue.size() != mStateVariablesFinalized.size())
             rValue.resize(mStateVariablesFinalized.size());
 
         noalias(rValue) = mStateVariablesFinalized;
     } else if (rThisVariable == CAUCHY_STRESS_VECTOR) {
-        if (rValue.size() != mStressVectorFinalized.size())
-            rValue.resize(mStressVectorFinalized.size());
-
-        noalias(rValue) = mStressVectorFinalized;
+        if (rValue.size() != TVoigtSize) rValue.resize(TVoigtSize);
```

Comment:
The `size` check is also carried out by the `resize` operation, so we don't need to do it here:
```suggestion
        rValue.resize(TVoigtSize);
```

Reply:
Done